### PR TITLE
Improvements to permutations.py by Ragnar

### PR DESCRIPTION
**Pull request by MurrayT**
Code style checked with
PEP8,
Radon, and
PyLint.
(Certain errors ignored).

### CODE REVIEW

File: permuta/misc/iterable_floor_and_ceiling.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        the i-th element of the iterable. The tuples are named tuples
+        with floor and ceiling attributes.
+    """
+    # TODO: Define behaviour for duplicate elements
+    dq = collections.deque()
+    smallest = None
+    biggest = None
+    index = 0
+    for element in iterable:
+        if index is 0:
```
Comment (SuprDewd) [inline permuta/misc/iterable_floor_and_ceiling.py]: Use **==** when comparing integers.

File: tests/test_permutation.py [hunk 2]
```diff
@@ -9,12 +9,12 @@ def test_init(self):
         with self.assertRaises(AssertionError): Permutation([2,1,2], check=True)
         with self.assertRaises(AssertionError): Permutation([1,1], check=True)
         with self.assertRaises(AssertionError): Permutation([2], check=True)
-        with self.assertRaises(AssertionError): Permutation(set([1,2,3]), check=True)
         with self.assertRaises(AssertionError): Permutation(5, check=True)
         with self.assertRaises(AssertionError): Permutation(None, check=True)
         Permutation([], check=True)
         Permutation([1], check=True)
         Permutation([4,1,3,2], check=True)
+        Permutation(set([1,2,3]), check=True)
```
Comment (SuprDewd) [inline tests/test_permutation.py]: It's a bit unfortunate that this works, since **set** is inherently unordered.

Comment (RagnarArdal) [inline tests/test_permutation.py]: Agreed, it is unfortunate; using a set to initialize a permutation is dumb. However, you should be able to do it. It is consistent with how tuples behave:
```
>>>tuple(set([1,2,3,4]))
(1, 2, 3, 4)
```
and we'll be making permutations a subclass of tuples in the near future.
At most we could have some sort of warning, but I think we are safe to assume some python aptitude from users of the permutation class.

File: permuta/misc/iterable_floor_and_ceiling.py (hunk 1 above)
Comment (RagnarArdal) [inline permuta/misc/iterable_floor_and_ceiling.py]: We were actually discussing this earlier!
Thanks for noticing, I'll get right on it.

**Comment by RagnarArdal** [comment]
I reckon this is all working now. You might want to hold off merging until you've modified the algorithms using permuta so that they don't use the 'features' of the earlier undefined permuta interface. E.g., passing a list to the avoids method (the argument must now be a permutation).

**Comment by MurrayT** [comment]
This is contained within the tuple branch.

**Pull request closed without merging**